app/web_framework/cluster_base_handler.py

```diff
@@ -21,9 +22,29 @@ class ClusterBaseHandler(tornado.web.RequestHandler):
         ItemNotFoundError: http.client.NOT_FOUND,
     }
 
-    def __init__(self, *args, **kwargs):
-        self._logger = log.get_logger(__name__)
-        super().__init__(*args, **kwargs)
+    def _handle_request_exception(self, ex):
+        """
+        This is the "catch-all" exception handler that Tornado uses to ensure exceptions are caught and a JSON response
+        is always returned to the client. (Tornado's default is an HTML response.) This also contains logic to map some
+        exception types to appropriate HTTP status codes. The default status code is 500 (generic server error).
+
+        :param ex: The exception that was caught
+        :type ex: Exception
+        """
+        # _handle_request_exception() is called in the exception handler, so we can still use logger.exception.
+        self._logger.exception('Exception occurred during request to {}.', self.request.uri)
+        status_code = self._exception_status_codes.get(type(ex), http.client.INTERNAL_SERVER_ERROR)
+        response = {'error': str(ex)}
+
+        self.set_status(status_code)
+        self.write(response)
+        self.finish()
+
+
+class ClusterBaseRequestHandler(ClusterBaseHandler):
```

Comment:
What is the difference between `ClusterBaseHandler` and `ClusterBaseRequestHandler`? I wonder if we can find better naming (or at least add class docs) to more clearly signify what should go in which class. They both handle requests, so it feels like those names are conceptually identical.

It looks like `ClusterBaseRequestHandler` is just for JSON API requests and `ClusterBaseHandler` is for any request, so what do you think of renaming `ClusterBaseRequestHandler` to maybe something like `ClusterBaseAPIHandler` or `ClusterBaseJsonHandler`?